src/register_target.cpp

```diff
@@ -57,7 +58,8 @@ target make_target(const std::string& name)
 {
     if(not contains(target_map(), name))
     {
-        auto target_name = make_shared_object_filename("migraphx_" + name);
+        auto target_name = make_shared_object_filename("migraphx_" + name) + "." +
+                           std::to_string(MIGRAPHX_SO_MAJOR_VERSION) + ".0";
```

Comment:
First you need to try and load it with the version appended and if that fails then try it without the version.